hypertension/dc-7101-algorithms/py-root/test/main_test.py

```diff
@@ -1,14 +1,214 @@
+                    "predominance_calculation": {
+                        "success": False,
+                    },
+                    "diastolic_history_calculation": {
+                        "diastolic_bp_predominantly_100_or_more": True,
+                        "success": True 
+                    },
+                    "requires_continuous_medication": {
+                        "continuous_medication_required": False,
+                        "success": True
+                    }
+                }
+            }
+        ),
+        # sufficiency and history algos fail
+        (
+            {
+                "bp": [
+                ],
+                "medication": [],
+                'date_of_claim': '2021-11-09',
+                'veteran_is_service_connected': True
+            },
+            {
+                "statusCode": 400,
+                "headers": {
+                    "Access-Control-Allow-Headers" : "Content-Type",
+                    "Access-Control-Allow-Origin": "*",
+                    "Access-Control-Allow-Methods": "OPTIONS,POST"
+                },
+                "body": {
+                    "predominance_calculation": {
+                        "success": False,
+                    },
+                    "diastolic_history_calculation": {
+                        "success": False 
+                    },
+                    "requires_continuous_medication": {
+                        "continuous_medication_required": False,
+                        "success": True
+                    }
+                }
+            }
+        ),
+        # Bad data (KeyError)
+        (
+            {
+                "bp": [
+                    {
+                        "diastolic": 111,
+                        "systolic": 200,
+                        "date": "2021-09-01"
+                    },
+                    {
+                        "medication": 120,
+                        "systolic": 180,
+                        "date": "2021-11-01"
+                    }
+                ],
+                "medication": [],
+                'date_of_claim': '2021-11-09',
+                'veteran_is_service_connected': True
+            },
+            {
+                "statusCode": 500,
+                "headers": {
+                    "Access-Control-Allow-Headers" : "Content-Type",
+                    "Access-Control-Allow-Origin": "*",
+                    "Access-Control-Allow-Methods": "OPTIONS,POST"
+                },
+                "body": {
+                    "predominance_calculation": {
+                        "success": False,
+                    },
+                    "diastolic_history_calculation": {
+                        "success": False 
+                    },
+                    "requires_continuous_medication": {
+                        "success": False
+                    }
+                }
+            }
+        ),
+        # Bad data (TypeError)
```

Comment:
I'm having a hard time understanding what's wrong here. Can we chat about it?

Reply:
This example has a string for the value of "diastolic" instead of an int; however, see the discussion about the 500 error above, it might be that we need to change/remove these tests

Reply:
add more descriptive comment